projects/plugins/jetpack/modules/google-fonts.php

```diff
@@ -104,4 +104,7 @@ function jetpack_add_google_fonts_provider() {
 
 add_filter( 'wp_resource_hints', '\Automattic\Jetpack\Fonts\Utils::font_source_resource_hint', 10, 2 );
 add_filter( 'pre_render_block', '\Automattic\Jetpack\Fonts\Introspectors\Blocks::enqueue_block_fonts', 10, 2 );
-add_action( 'init', '\Automattic\Jetpack\Fonts\Introspectors\Global_Styles::enqueue_global_styles_fonts' );
+// The priority for the next hook is is set to 22 because the default (or lower)
+// because it was causing style issues if it ran before the `gutenberg_register_webfonts_from_theme_json`
+// which runs in the same hook but at priority 21.
```

Comment:
I did some more investigation. The real issue is that it was running in between Gutenberg's `gutenberg_reregister_core_block_types` (which deregisters the core styles) at the default priority 10 and the re-registrations of the Gutenberg replacements that happen at priority 20.

`gutenberg_register_webfonts_from_theme_json` doesn't seem to matter with respect to the missing style issue. Although in Gutenberg 13.4.0 and earlier, that hook running before the deregistration prevented our hook running after from being a problem.

```suggestion
// The priority for the next hook is is set to 22 because it needs to run after Gutenberg's
// re-registration (at priority 22) of the core blocks it de-registers (at the default priority 10),
// otherwise Gutenberg caches an incorrect state.
```

Also FWIW priority 21 also seems to work. Verified that on a JN site.